src/polyfem/State.cpp

```diff
@@ -1102,66 +1104,115 @@ namespace polyfem
 		Eigen::MatrixXd node_positions;
 		Eigen::MatrixXi boundary_edges, boundary_triangles;
 		std::vector<Eigen::Triplet<double>> displacement_map_entries;
-		io::OutGeometryData::extract_boundary_mesh(*mesh, n_bases, bases, total_local_boundary,
-												   node_positions, boundary_edges, boundary_triangles, displacement_map_entries);
+		io::OutGeometryData::extract_boundary_mesh(
+			*mesh, n_bases, bases, total_local_boundary, node_positions,
+			boundary_edges, boundary_triangles, displacement_map_entries);
+		// assert(node_positions.rows() == n_bases);
 
-		Eigen::VectorXi codimensional_nodes;
-		if (obstacle.n_vertices() > 0)
-		{
-			// n_bases already contains the obstacle vertices
-			const int n_v = n_bases - obstacle.n_vertices();
+		// n_bases already contains the obstacle vertices
+		const int num_fe_nodes = node_positions.rows() - obstacle.n_vertices();
 
-			if (obstacle.v().size())
-				node_positions.block(n_v, 0, obstacle.v().rows(), obstacle.v().cols()) = obstacle.v();
+		Eigen::MatrixXd collision_vertices;
+		Eigen::VectorXi collision_codim_vids;
+		Eigen::MatrixXi collision_edges, collision_triangles;
 
-			if (!displacement_map_entries.empty())
+		if (args.value("/contact/collision_mesh/enabled"_json_pointer, false))
+		{
+			const json collision_mesh_args = args["/contact/collision_mesh"_json_pointer];
+			assert(displacement_map_entries.empty());
+
+			Eigen::MatrixXi codim_edges;
+			mesh::read_surface_mesh(
+				resolve_input_path(collision_mesh_args["mesh"]),
+				collision_vertices, collision_codim_vids, codim_edges, collision_triangles);
+
+			if (collision_triangles.size())
+				igl::edges(collision_triangles, collision_edges);
+
+			append_rows(collision_edges, codim_edges);
+
+			// TODO: transform the collision mesh in the same way the rest of the mesh is transformed
+			// V = V * T.transpose();
+
+			H5Easy::File file(resolve_input_path(collision_mesh_args["linear_map"]));
```

Comment:
we wanted to switch hdf5 library :(

Reply:
maybe use save market from eigen?

Reply:
Do you mean load market?

Reply:
[h5pp](https://github.com/wildmeshing/h5pp) supports Eigen too. I can switch to that library.

Reply:
we need to pull the new version of paraviewo